### PR TITLE
Send order data to database on submit

### PR DESCRIPTION
**Pull request by khushi02**

### CODE REVIEW

**Review by McSobkov — APPROVED** [review]
Looks good to me! You may want to remove the commented-out console.logs in Order.tsx, but if they may be helpful later, that's completely valid.

**Comment by SeanEZhou** [comment]
Hi Khushi when I tested it out locally I don’t actually see the database got updated

File: dronuts-frontend/src/pages/Order.tsx [hunk 1]
```diff
@@ -1,18 +1,146 @@
...
-  return <Checkout />;
+
+  // console.log(cart);
+  // console.log(address);
+ 
+  function Copyright() {
+    return (
+      <Typography variant="body2" color="text.secondary" align="center">
+        {'Copyright © '}
+        <Link color="inherit" href="https://mui.com/">
```
Comment (phi-h-nguyen) [inline dronuts-frontend/src/pages/Order.tsx:35]: I think we can change this link to the dronuts menu page

File: dronuts-frontend/src/pages/Order.tsx [hunk 2]
```diff
@@ -1,18 +1,146 @@
...
+import AddressForm from '../components/AddressForm';
+import TimeEstimate from '../components/TimeEstimate';
+import PaymentForm from '../components/PaymentForm';
+import Review from '../components/Review';
+import Header from "../components/Header";
 import "./Order.css";
 import { useLocation } from 'react-router-dom'
 
-
 function App() {
```
Comment (phi-h-nguyen) [inline dronuts-frontend/src/pages/Order.tsx:24]: I think function names for pages/components should be consistent with the file name

File: dronuts-frontend/src/pages/Order.tsx [hunk 3]
```diff
@@ -1,18 +1,146 @@
...
+
+    const handleSubmit = () => {
+      handleNext();
+
+      const data = {'customerId': 100*(Math.random()),
+      'items': cart,
+      'address': '5000 Forbes Ave'};
+
+      try {
+        const sendData = axios.post('/order', data).then((res) => (res.json()));
```
Comment (phi-h-nguyen) [inline dronuts-frontend/src/pages/Order.tsx:81]: Is the axios library already in in the development dependancies?

File: dronuts-frontend/src/pages/Order.tsx [hunk 4]
```diff
@@ -1,18 +1,146 @@
...
+              ))}
+            </Stepper>
+            <React.Fragment>
+              {activeStep === steps.length ? (
+                <React.Fragment>
+                  <Typography variant="h5" gutterBottom>
+                    Thank you for your order.
+                  </Typography>
+                  <Typography variant="subtitle1">
+                    Your order number is #2001539. We have emailed your order
```
Comment (phi-h-nguyen) [inline dronuts-frontend/src/pages/Order.tsx:112]: Are we able to update the order number based on the backend?

Comment (phi-h-nguyen) [reply dronuts-frontend/src/pages/Order.tsx:112]: And with the shipping address

File: dronuts-frontend/src/pages/Order.tsx [hunk 5]
```diff
@@ -1,18 +1,146 @@
...
+    const handleBack = () => {
+      setActiveStep(activeStep - 1);
+    };
+
+    const handleSubmit = () => {
+      handleNext();
+
+      const data = {'customerId': 100*(Math.random()),
+      'items': cart,
+      'address': '5000 Forbes Ave'};
```
Comment (phi-h-nguyen) [inline dronuts-frontend/src/pages/Order.tsx:78]: I think we can use the actual address from the form instead of the placeholder

**Review by phi-h-nguyen — COMMENTED** [review]
Overall looks good! Just added a comment about react function names and some questions about the backend communication

File: dronuts-frontend/src/pages/Order.tsx (hunk 2 above)
Comment (khushi02) [reply dronuts-frontend/src/pages/Order.tsx:24]: I could change it, but we actually use App on all the other pages, so I wanted to keep it consistent with that.

**Comment by khushi02** [comment]
Thanks for the comments! The axios dependency is fixed, I forgot to push the most recent version with it included which is why there was an error with the database when running. @phi-h-nguyen, for the shipping address and order number, there are issues with accessing cart data in the review component, so I was aiming to fix that issue next sprint.

**Comment by SeanEZhou** [comment]
Hi Khushi now I get the following error when I run the frontend via “nom start”:

Module not found: Error: Can't resolve 'axios' in '/Users/seanzer0/Desktop/dronuts2022-group_1_spring_2022/dronuts-frontend/src/pages'

**Comment by khushi02** [comment]
Hey Sean, It's working on my end. I would make sure you have the axios import statement and all necessary dependencies installed.

**Comment by SeanEZhou** [comment]
Cool, error’s gone after installing axios. In that case Khushi can you run “npm install axios --save” to save the dependency to our package.json?

**Comment by SeanEZhou** [comment]
But when I checkout an order it still doesn’t show up in our database?

**Pull request merged**